src/components/About/WarningBanner.js

```diff
@@ -0,0 +1,83 @@
+import _ from 'lodash';
+import React, { useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import {
+  Headline,
+  List,
+  MessageBanner,
+} from '@folio/stripes-components';
+
+import { isVersionCompatible } from '../../discoverServices';
+import css from './About.css';
+
+const WarningBanner = ({
+  interfaces,
+  modules,
+}) => {
+  const allInterfaces = useMemo(() => {
+    const modulesArray = _.flatten(_.values(modules));
+
+    return modulesArray.reduce((prev, curr) => Object.assign(prev, curr.okapiInterfaces), {});
+  }, [modules]);
+
+const missingModules = useMemo(
+  () => Object.entries(allInterfaces)
+    .filter(([key]) => !(key in interfaces))
+    .map(([key, value]) => `${key} ${value}`),
+  [allInterfaces, interfaces]
+);
+
+  const incompatibleModules = useMemo(() => {
+    return Object.keys(allInterfaces).reduce((prev, curr) => {
+      if (interfaces[curr] && !isVersionCompatible(interfaces[curr], allInterfaces[curr])) {
+        return [...prev, `${curr} ${allInterfaces[curr]}`];
+      } else {
+        return prev;
+      }
+    }, []);
+  }, [allInterfaces, interfaces]);
+
+  const missingModulesCount = useMemo(() => missingModules.length, [missingModules]);
+
+  const incompatibleModulesCount = useMemo(() => incompatibleModules.length, [incompatibleModules]);
```

Comment:
I think now, without `compact`, useMemo() is excessive here. No need to memoize just an array length.